### PR TITLE
[RFC][Behat] Promotion system feature files

### PR DESCRIPTION
**Pull request by michalmarcinkowski**
Feature files with scenarios for:
* applying shipping percentage discount.
* applying promotion based on cart quantity
* applying promotion based on items total
* reapplying promotion on cart change

### CODE REVIEW

File: features/promotion/applying_promotion_based_on_cart_quantity.feature [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  As a Visitor
+  I want to have promotion applied to my cart when my cart quantity qualifies
+
+  Background:
+    Given the store is operating on a single "France" channel
+    And the store has a product "PHP T-Shirt" priced at "€100.00"
+    And there is a promotion "Holiday promotion"
+
+  @todo
+  Scenario: Applying discount when cart quantity is above the quantity from criteria and should be above
```
Comment (michalmarcinkowski) [inline features/promotion/applying_promotion_based_on_cart_quantity.feature]: Receiving discount when buying more than the required quantity

File: features/promotion/applying_promotion_based_on_cart_quantity.feature [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    Given the promotion gives "€10.00" fixed discount to every cart with quantity above 2
+    When I add 3 products "PHP T-Shirt" to the cart
+    Then my cart total should be "€290.00"
+    And my cart promotions should be "-€10.00"
+
+  @todo
+  Scenario: Not applying discount when cart quantity is equal to the amount from criteria but should be above
```
Comment (michalmarcinkowski) [inline features/promotion/applying_promotion_based_on_cart_quantity.feature]: Not receiving discount when buying less than required quantity

File: features/promotion/applying_promotion_based_on_cart_quantity.feature [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  @todo
+  Scenario: Not applying discount when cart quantity is lesser than the amount from criteria but should be above
+    Given the promotion gives "€10.00" fixed discount to every cart with quantity above 2
+    When I add product "PHP T-Shirt" to the cart
+    Then my cart total should be "€100.00"
+    And my cart promotions should be "€0.00"
+
+  @todo
+  Scenario: Applying discount on cart with multiple products when cart quantity is above the quantity from criteria
```
Comment (michalmarcinkowski) [inline features/promotion/applying_promotion_based_on_cart_quantity.feature]: Receiving discount when buying different products with the required quantity

File: features/promotion/applying_promotion_based_on_items_total.feature [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  As a Visitor
+  I want to have promotion applied when my items total is qualified
+
+  Background:
+    Given the store is operating on a single "France" channel
+    And the store has a product "PHP T-Shirt" priced at "€100.00"
+    And there is a promotion "Holiday promotion"
+
+  @todo
+  Scenario: Applying discount when items total is above the amount from criteria and should be above
```
Comment (michalmarcinkowski) [inline features/promotion/applying_promotion_based_on_items_total.feature]: Receiving discount when buying items for more than required total value

File: features/promotion/reapplying_promotion_on_cart_change.feature [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  I want to have proper discount applied after every operation on my cart
+
+  Background:
+    Given the store is operating on a single "France" channel
+    And the store has a product "PHP T-Shirt" priced at "€100.00"
+    And there is a promotion "Holiday promotion"
+    And I am logged in customer
+
+  @todo
+  Scenario: Not applying promotion on shipping on empty cart after product removal
```
Comment (pjedrzejewski) [inline features/promotion/reapplying_promotion_on_cart_change.feature]: Not receiving discount on shipping after removing last item from cart

File: features/promotion/reapplying_promotion_on_cart_change.feature [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    And the promotion gives "100%" percentage discount on shipping to every order
+    And I have product "PHP T-Shirt" in the cart
+    When I proceed selecting "DHL" shipping method
+    And I remove product "PHP T-Shirt" from the cart
+    Then my cart total should be "€0.00"
+    And my cart shipping fee should be "€0.00"
+    And my cart promotions should be "€0.00"
+
+  @todo
+  Scenario: Reapplying promotion on shipping after shipping method change
```
Comment (pjedrzejewski) [inline features/promotion/reapplying_promotion_on_cart_change.feature]: Receiving discount after changing the shipping method

File: features/promotion/reapplying_promotion_on_cart_change.feature [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    And the promotion gives "100%" percentage discount on shipping to every order
+    And I have product "PHP T-Shirt" in the cart
+    And I chose "DHL" shipping method
+    When I change shipping method to "FedEx"
+    Then my cart total should be "€100.00"
+    And my cart shipping fee should be "€30.00"
+    And my cart promotions should be "-€30.00"
+
+  @todo
+  Scenario: Reapplying promotion when new product added to cart after all products removal
```
Comment (pjedrzejewski) [inline features/promotion/reapplying_promotion_on_cart_change.feature]: Receiving discount after removing an item from the cart and then adding another one

File: features/promotion/reapplying_promotion_on_cart_change.feature [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  @todo
+  Scenario: Not applying promotion on shipping on empty cart after product removal
+    Given the store has "DHL" shipping method with "€10.00" fee
+    And the promotion gives "100%" percentage discount on shipping to every order
+    And I have product "PHP T-Shirt" in the cart
+    When I proceed selecting "DHL" shipping method
+    And I remove product "PHP T-Shirt" from the cart
+    Then my cart total should be "€0.00"
+    And my cart shipping fee should be "€0.00"
+    And my cart promotions should be "€0.00"
```
Comment (pjedrzejewski) [inline features/promotion/reapplying_promotion_on_cart_change.feature]: And my discount should be
And discount on my cart should be

I think I prefer first one.

File: features/promotion/reapplying_promotion_on_cart_change.feature [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    Given the store has a product "Symfony T-Shirt" priced at "€150.00"
+    And the promotion gives "€10.00" fixed discount to every order
+    And I had product "PHP T-Shirt" in the cart
+    But I removed this product from the cart
+    When I add product "Symfony T-Shirt" to the cart
+    Then my cart total should be "€140.00"
+    And my cart promotions should be "-€10.00"
+
+  @todo
+  Scenario: Not applying discount on cart after items total is decreased below the amount from criteria
```
Comment (pjedrzejewski) [inline features/promotion/reapplying_promotion_on_cart_change.feature]: Not receiving a discount when cart does not meet the required total value after removing an item

File: features/promotion/reapplying_promotion_on_cart_change.feature [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    Given the promotion gives "€10.00" fixed discount to every cart with items total at least "€120.00"
+    And I have 2 products "PHP T-Shirt" in the cart
+    When I change "PHP T-Shirt" quantity to 1
+    Then my cart total should be "€100.00"
+    And my cart promotions should be "€0.00"
+
+  @todo
+  Scenario: Not applying discount on cart after cart quantity is decreased below the amount from criteria
```
Comment (pjedrzejewski) [inline features/promotion/reapplying_promotion_on_cart_change.feature]: Not receiving a discount when cart does not meet the required quantity after removing an item

**Pull request merged**